### PR TITLE
EZP-31127: Skipped non-existing Content Types in ContentTypeIdentifier criterion

### PR DESCRIPTION
**Pull request by emodric**
> https://jira.ez.no/browse/EZP-31127

Variant of https://github.com/ezsystems/ezpublish-kernel/pull/2872 PR for Solr

### CODE REVIEW

File: lib/Query/Common/CriterionVisitor/ContentTypeIdentifierIn.php [hunk 1]
```diff
@@ -66,16 +67,27 @@ public function visit(Criterion $criterion, CriterionVisitor $subVisitor = null)
...
-                array_map(
-                    function ($value) use ($contentTypeHandler) {
-                        return 'content_type_id_id:"' . $contentTypeHandler->loadByIdentifier($value)->id . '"';
-                    },
-                    $criterion->value
-                )
-            ) .
-            ')';
+        $idQueries = array_map(
+            function ($id) {
```
Comment (andrerom) [inline lib/Query/Common/CriterionVisitor/ContentTypeIdentifierIn.php]:
```suggestion
            static function ($id) {
```

**Comment by andrerom** [comment]
Will you add warning logging here as well like kernel?

**Comment by emodric** [comment]
@andrerom Done.

**Comment by emodric** [comment]
@micszo I'll look into it.

**Comment by emodric** [comment]
@micszo I've pushed the fix. Please test again!

**Comment by lserwatka** [comment]
@alongosz could you merge it up?

**Comment by alongosz** [comment]
Merged up to 2.0 as 975e3c8 and to master as  f2f2c7b.

**Pull request merged**